troposphere/static/js/components/admin/ResourceRequest/ResourceRequestView.jsx

```diff
@@ -96,68 +45,116 @@ export default React.createClass({
         }
     },
 
+    renderAllocationsSection() {
+        let { allocationSources } = this.props;
+        let { section } = this.style();
+
+        let body = <p>There are no allocations</p>;
+        if (allocationSources) {
+            body = allocationSources.map(allocationSource => {
+                let id = allocationSource.get("id");
+                let props = {
+                    onSave: this.props.onAllocationSave,
+                    allocationSource,
+                }
+                return <AllocationSourceView key={id} { ...props }/>;
+            })
+        }
+
+        return (
+        <div style={section}>
+            { body }
+        </div>
+        );
+    },
+
+    renderIdentitiesSection() {
+        let { identities, onIdentitySave: onSave } = this.props;
+        let { section } = this.style();
```

Comment:
Just want to note for folks _watching_ (and for me):
 ... the statement that appears line 72 here is I think [a variation of destructuring they call "unpacking"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) where the value of `onIdentitySave` from `this.props` is _bound_ to `onSave`.

You'll see that `onSave` is later used in the `.map(...)`.